### ISSUE TITLE
Destructured parameter renamed

### ISSUE THREAD

**Issue by marian-r** [babel/babel#1566]
Note the following ES6 code with a destructured parameter:
```js
var f = function ({ foo = 'bar' }) {
  
  console.log(foo);
  
  var obj = {
    // same name as parameter
    foo: function () {
      console.log(foo);
    }
  };
};
```
Babel produces the following code:
```js
'use strict';

var f = function f(_ref) {
  var _ref$_foo = _ref._foo;

  var _foo = _ref$_foo === undefined ? 'bar' : _ref$_foo;

  console.log(_foo);

  var obj = {
    // same name as parameter
    foo: function foo() {
      console.log(_foo);
    }
  };
};
```
Parameter `foo` is renamed as a local variable `_foo` due to the function `obj.foo` with the same name. But the problem is that the destructured parameter is renamed too - `_ref._foo`. So when you call the function
```js
f({ foo: 'baz' });
```
value of `_ref._foo` is obviously undefined thus `_foo` has default value 'bar' instead of 'baz'.

This issue can be fixed by not renaming the parameter when destructuring:
```js
var _ref$_foo = _ref.foo;
var _foo = _ref$_foo === undefined ? 'bar' : _ref$_foo;
```
I can contribute with the tests for this issue if it would be helpful, or you can use my code examples in the tests.

**Comment by sebmck** [comment]
Sorry for the delay, fixed as of 5.4.4. Thanks!

**Comment by marian-r** [comment]
Thank you, you fixed it really quickly :)
